portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,51 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import com.google.sps.data.LoginAccount;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+    protected UserService userService;
+    protected Gson gson;
+
+    public LoginServlet() {
+        super();
+
+        userService = UserServiceFactory.getUserService();
+        gson = new Gson();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("text/html");
+        boolean isLoggedIn = userService.isUserLoggedIn();
+
+        String urlToRedirectToAfterUserLogsOut = "/index.html";
+        String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+        String urlToRedirectToAfterUserLogsIn = "/index.html";
+        String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+
+        LoginAccount account = new LoginAccount(isLoggedIn, loginUrl, logoutUrl);
```

Comment:
it's OK to just do

`String logoutUrl = userService.createLogoutURL("/index.html");`

since you're not using the variable and it's clear what's going on.

You would even be OK to just do

`LoginAccount account = new LoginAccount(isLoggedIn, userService.createLoginURL("/index.html"), userService.createLogoutURL("/index.html"));`

If you wanted to not have the literal strings, you could always add them as constants in the class since they don't change between requests.

```
public class LoginServlet extends HttpServlet {
  protected static final String LOGIN_REDIRECT = "index.html";
  protected static final String LOGOUT_REDIRECT = "index.html";

...

LoginAccount account = new LoginAccount(isLoggedIn, userService.createLoginURL(LOGIN_REDIRECT), userService.createLogoutURL(LOGOUT_REDIRECT));

```

Constants in Java are always in all caps because I don't know why, but they are.  Static in the declaration means that it's OK for the compiler to reuse the same variable across multiple instantiations of the class since it will never change, so if you had 2 LoginServlet classes for whatever reason, it could save space by only bothering to keep track of one value of LOGIN_REDIRECT or just replacing it with the string everywhere in the file and not make a variable in the first place.